src/main.rs

```diff
@@ -0,0 +1,401 @@
+use async_trait::async_trait;
+use clap::Parser;
+use core::time::Duration;
+use log::error;
+use reqwest::{header::CONTENT_TYPE, Url};
+use reqwest_middleware::{ClientBuilder, ClientWithMiddleware};
+use reqwest_retry::{policies::ExponentialBackoff, RetryTransientMiddleware};
+use scroll_proving_sdk::{
+    config::{CloudProverConfig, Config},
+    prover::{
+        proving_service::{
+            GetVkRequest, GetVkResponse, ProveRequest, ProveResponse, QueryTaskRequest,
+            QueryTaskResponse, TaskStatus,
+        },
+        types::CircuitType,
+        ProverBuilder, ProvingService,
+    },
+    utils::init_tracing,
+};
+
+use chrono::{DateTime, NaiveDateTime, Utc};
+use serde::{Deserialize, Deserializer, Serialize};
+
+/// API version used by the Snarkify platform.
+const API_VERSION: &'static str = "v1";
+
+fn deserialize_datetime<'de, D>(deserializer: D) -> Result<Option<DateTime<Utc>>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    // The datetimes from the Snarkify API does not provide timezone information,
+    // so we assume it is UTC.
+    let s: Option<String> = Option::deserialize(deserializer)?;
+    s.map(|s| {
+        NaiveDateTime::parse_from_str(&s, "%Y-%m-%dT%H:%M:%S")
+            .map(|dt| DateTime::<Utc>::from_naive_utc_and_offset(dt, Utc))
+            .map_err(serde::de::Error::custom)
+    })
+    .transpose()
+}
+
+#[derive(Parser, Debug)]
+#[clap(disable_version_flag = true)]
+struct Args {
+    /// Path to the configuration file in JSON format.
+    /// Please refer to config.json in https://github.com/snarkify/scroll-proving-sdk
+    #[arg(long = "config", default_value = "config.json")]
+    config_file: String,
+    /// Unique UUID for the service in Snarkify platform.
+    #[arg(long = "service-id")]
+    service_id: String,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct SnarkifyGetTaskResponse {
+    /// Task UUID in Snarkify platform.
+    pub task_id: String,
+    #[serde(deserialize_with = "deserialize_datetime")]
+    pub created: Option<DateTime<Utc>>,
+    #[serde(deserialize_with = "deserialize_datetime")]
+    pub started: Option<DateTime<Utc>>,
+    #[serde(deserialize_with = "deserialize_datetime")]
+    pub finished: Option<DateTime<Utc>>,
+    pub state: SnarkifyTaskState,
+    /// Task input data necessary for the proof generation.
+    pub input: String,
+    /// Task proof.
+    pub proof: Option<String>,
+    /// Task error message.
+    pub error: Option<String>,
```

Comment:
Such rustdoc are not useful and can be removed. 

However, normal rustdocs can be very useful. For example, "serialized in hex binary proof" could simplify parsing of this code by an order of magnitude

Reply:
sg!